### PR TITLE
Using `dvc.yaml` instead of `artifacts.yaml`

### PR DESCRIPTION
**Pull request by aguschin**
implements part of https://github.com/iterative/dvc.org/issues/4423

https://dvc-org-dvc-artifacts-8epucpc9.herokuapp.com/doc

### CODE REVIEW

File: content/docs/dvclive/index.md
Comment (daavoo) [inline content/docs/dvclive/index.md:53]: Let's keep the alphabetical order.

**Review by daavoo — COMMENTED** [review]
Changes look good to me and they technically document all parts.

I am missing some sort of narrative tying things together, though.

Not sure where it would fit*, but it feels that even after these changes it would be hard for someone entering the DVC docs to find out about / understand the model registry.

---

*According to the metrics in plausible, people usually flow from `root->get started->user guide/cmd ref` in a decreasing number of views.

File: content/docs/start/experiments/experiment-tracking.md
Comment (dberenbaum) [inline content/docs/start/experiments/experiment-tracking.md:43]: Do we need to pass `DVCLiveLogger(save_dvc_exp=True, experiment=live)` here (and similar for the other examples)?

File: content/docs/dvclive/index.md
Comment (dberenbaum) [inline content/docs/dvclive/index.md:46]: Do we need a separate tab? Why not just update the `Artifacts` tab? It's the first one we show, so there will still be an obvious example of logging a model.

File: content/docs/start/experiments/experiment-tracking.md
Comment (dberenbaum) [inline content/docs/start/experiments/experiment-tracking.md:47]: Not sure this pattern is that helpful for model registry since it will save the full directory of checkpoints. WDYT about something like this:

```
with Live(save_dvc_exp=True) as live:
    checkpoint = pl.callbacks.ModelCheckpoint(dirpath="mymodel")
    trainer = Trainer(
        logger=DVCLiveLogger(save_dvc_exp=True), callbacks=checkpoint
    )
    trainer.fit(model)
    live.log_artifact(checkpoint.best_model_path, type="model")
```

File: content/docs/start/experiments/experiment-tracking.md
Comment (dberenbaum) [inline content/docs/start/experiments/experiment-tracking.md:41]: Let's also add `log_artifact` to the examples in https://dvc.org/doc/dvclive/ml-frameworks.

File: content/docs/start/experiments/experiment-tracking.md
Comment (dberenbaum) [inline content/docs/start/experiments/experiment-tracking.md:101]: Feels like we need one sentence at the end of this section (below these examples) to explain log_artifact (saves artifacts with dvc and enables managing them with model registry).

File: content/docs/user-guide/experiment-management/index.md
Comment (dberenbaum) [inline content/docs/user-guide/experiment-management/index.md:49]: Minor suggestion/not blocking: move it below metrics, plots, and parameters. I think those are still the first thing people want to know about when they start experiment tracking.

Comment (dberenbaum) [reply content/docs/user-guide/experiment-management/index.md:49]: Another minor/non-blocking suggestion: Datasets and models -> Models and datasets.

File: content/docs/dvclive/index.md
Comment (dberenbaum) [inline content/docs/dvclive/index.md:49]: No strong opinion, but I guess you decided not to suggest explicit names?

File: content/docs/studio/user-guide/model-registry/add-a-model.md
Comment (dberenbaum) [inline content/docs/studio/user-guide/model-registry/add-a-model.md:4]:
```suggestion
your model registry, Iterative Studio creates an annotation for it in a
```

File: content/docs/studio/user-guide/model-registry/add-a-model.md
Comment (dberenbaum) [inline content/docs/studio/user-guide/model-registry/add-a-model.md:32]: Do we need this much text about what path to use? Seems like the options are either:
1. If the model file is in the Git repository (including if it is saved with DVC and/or MLEM), enter the relative path of the model (from the repository root).
2. Otherwise, enter the URL to the model file in the cloud.

**Review by dberenbaum — DISMISSED** [review]
Thanks @aguschin!

In addition to the inline comments, there's a mention of `artifacts.yaml` in https://dvc.org/doc/studio/get-started#manage-models.

I think we can address those comments and merge, but I wouldn't close #4423 because I think we need to make bigger changes how we explain the model registry throughout the docs, so let's discuss in #4423.

File: content/docs/start/experiments/experiment-tracking.md
Comment (daavoo) [reply content/docs/start/experiments/experiment-tracking.md:43]: Yes, I didn´t notice.

File: content/docs/start/experiments/experiment-tracking.md
Comment (daavoo) [reply content/docs/start/experiments/experiment-tracking.md:47]: If we do this for lightning, we should probably also update the setup for the other tabs so the best model is the one logged instead of the latest

File: content/docs/dvclive/index.md
Comment (aguschin) [reply content/docs/dvclive/index.md:40]: Right, missed that. Let's suggest it.

File: content/docs/studio/user-guide/model-registry/add-a-model.md
Comment (aguschin) [reply content/docs/studio/user-guide/model-registry/add-a-model.md:9]: Yes, we should. Added that to your comment.

File: content/docs/user-guide/project-structure/dvcyaml-files.md
Comment (aguschin) [reply content/docs/user-guide/project-structure/dvcyaml-files.md:21]: I guess we can keep this and update later. Added to your comment.

File: content/docs/start/experiments/experiment-tracking.md
Comment (aguschin) [reply content/docs/start/experiments/experiment-tracking.md:47]: > we should probably also update the setup for the other tabs

Not quite get for which tabs exactly. For HF we save best iteration explicitly IIUC, for General Python API this is irrelevant, and for keras I'm also not sure we're saving each iteration. Can you please explain @daavoo ?

File: content/docs/start/experiments/experiment-tracking.md
Comment (aguschin) [reply content/docs/start/experiments/experiment-tracking.md:101]: edited paragraph few lines below to explain this.

File: content/docs/start/experiments/experiment-tracking.md
Comment (aguschin) [reply content/docs/start/experiments/experiment-tracking.md:43]: updated, please check!

File: content/docs/start/experiments/experiment-tracking.md
Comment (aguschin) [reply content/docs/start/experiments/experiment-tracking.md:41]: This will take a while since I'm not that familiar with each of them. Adding this to your comment, and let's do this in a follow-up PR.

File: content/docs/start/experiments/experiment-tracking.md
Comment (dberenbaum) [inline content/docs/start/experiments/experiment-tracking.md:127]:
```suggestion
[track your model with DVC](/doc/dvclive/live/log-artifact) and enable managing it with
```

The current link just goes to a similar code snippet that then refers to /doc/dvclive/live/log-artifact.

File: content/docs/studio/user-guide/model-registry/add-a-model.md
Comment (dberenbaum) [reply content/docs/studio/user-guide/model-registry/add-a-model.md:32]: @aguschin Thoughts on this?

Comment (aguschin) [reply content/docs/studio/user-guide/model-registry/add-a-model.md:32]: Looks right to me. Fixing!

**Comment by aguschin** [comment]
@dberenbaum Please note I enabled auto-merge to merge this once you approve. Thank you and @daavoo for the review!

**Review by dberenbaum — APPROVED** [review]

**Pull request merged**
